### PR TITLE
Remove duplicate tags before making API calls

### PR DESCRIPTION
**Pull request by Xenwar**
Remove duplicate tags before making API calls.

Tags that come from instance and port levels could contain duplicate values, causing failure when making api call.
This PR removes duplicate values before making the call. The change is done on the `replaceAllAttributesTags` function in order to avoid making changes in multiple places where tags are collected.

fixes https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1111

### CODE REVIEW

**Comment by Xenwar** [comment]
Note:  [WIP] is added to prevent merge while the manifests contain test data

**Comment by iamemilio** [comment]
seems reasonable
/lgtm

File: hack/ci/cloud-init/controller.yaml.tpl [hunk 1]
```diff
@@ -32,8 +32,11 @@
     # Placement service needed for Nova
     ENABLED_SERVICES+=,placement-api,placement-client
     # Glance
-    ENABLED_SERVICES+=,g-api
+    ENABLED_SERVICES+=,g-api,g-reg
 
+    # Octavia-Neutron
+    Q_AGENT=linuxbridge
+    ENABLED_SERVICES+=,neutron-api,neutron-agent,neutron-dhcp,neutron-l3,neutron-trunk
```
Comment (jichenjc) [inline hack/ci/cloud-init/controller.yaml.tpl:39]: what's this for? seems not related?

**Comment by Xenwar** [comment]
Test passed. 
Changes done for testing purpose are removed.
This PR can be reviewed with just the required changes to fix the duplicate tags problem.

File: pkg/cloud/services/networking/service.go [hunk 2]
```diff
@@ -89,15 +89,25 @@ func (s *Service) replaceAllAttributesTags(eventObject runtime.Object, resourceT
 		record.Warnf(eventObject, "FailedReplaceAllAttributesTags", "Invalid resourceType argument in function call")
 		panic(fmt.Errorf("invalid argument: resourceType, %s, does not match allowed arguments: %s or %s", resourceType, trunkResource, portResource))
 	}
+	// remove duplicate values from tags
```
Comment (jichenjc) [inline pkg/cloud/services/networking/service.go:92]: as follow up ,we can create some ut (to be honest we don't have too much so not a mandatory thing for this PR )
to test this works ..

Comment (Xenwar) [reply pkg/cloud/services/networking/service.go:92]: This one is more of a user's mistake that gives duplicate tags, not a use case. 
So, we may not need a test case for it. 
However, the previous test was run to validate the code: see https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1112#issuecomment-1012206109

Comment (jichenjc) [reply pkg/cloud/services/networking/service.go:92]: you are right :)
thanks for the great info :~

**Comment by Xenwar** [comment]
For future reference, here is an entry from the capo log for the above input. 
`I0114 06:43:46.892107       1 recorder.go:103] events "msg"="Normal"  "message"="Replaced all attributestags for 9da95b85-fffb-41f7-9c8c-633f3060dc54 with tags [tag1 tag2 tag3]" "object"={"kind":"OpenStackMachine","namespace":"e2e-hphugc","name":"cluster-e2e-hphugc-control-plane-mvwx4","uid":"96a075e6-747b-4ef9-bf74-07489fb28b03","apiVersion":"infrastructure.cluster.x-k8s.io/v1beta1","resourceVersion":"3794"} "reason"="SuccessfulReplaceAllAttributeTags"
`
